# **Libraries**

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

# **Load The Dataset**

In [2]:
df = pd.read_csv(r'C:\Users\Samuel\Documents\Projects\marketing_campaign_analysis\dataset\digital_marketing_campaign_dataset.csv')
df.sample(10)

CustomerID  Age  Gender  Income CampaignChannel   CampaignType  \
721         8721   37  Female   32588        Referral      Retention   
4564       12564   53    Male  106290             PPC      Retention   
4397       12397   49  Female   97391             PPC  Consideration   
4748       12748   27    Male  103959    Social Media      Awareness   
7703       15703   28    Male  144887        Referral  Consideration   
3131       11131   21  Female   64035        Referral      Retention   
4509       12509   48    Male   95525             PPC      Awareness   
1441        9441   36    Male   53149        Referral  Consideration   
6249       14249   59  Female   68339             PPC      Awareness   
2988       10988   37  Female   96942    Social Media      Retention   

          AdSpend  ClickThroughRate  ConversionRate  WebsiteVisits  \
721   2367.356530          0.237022        0.173390             48   
4564  6919.232083          0.012299        0.069628              2   
4397  7088.408130          0.253899        0.086032             29   
4748  9899.837876          0.080511        0.165916             17   
7703  7781.893988          0.036420        0.093683             39   
3131  1709.646502          0.088902        0.115995             32   
4509   972.603138          0.063942        0.171695              3   
1441  4308.295140          0.034293        0.194024             49   
6249  2396.434107          0.166969        0.154243              6   
2988  5269.080006          0.131998        0.145011             37   

      PagesPerVisit  TimeOnSite  SocialShares  EmailOpens  EmailClicks  \
721        4.761650    0.707391             9           7            6   
4564       3.956321   11.313252            91          16            3   
4397       1.529574    9.807894            96          18            8   
4748       5.832874    7.503303            27           3            4   
7703       9.047662    6.693311            36           6            9   
3131       5.357372    0.581506             3          17            7   
4509       8.307292   14.889766            95           6            5   
1441       9.623273    7.351469            44           2            4   
6249       8.250205   13.322380            79           0            6   
2988       4.718965    7.797316            85           2            7   

      PreviousPurchases  LoyaltyPoints AdvertisingPlatform AdvertisingTool  \
721                   3           2596            IsConfid      ToolConfid   
4564                  4           3545            IsConfid      ToolConfid   
4397                  8           1463            IsConfid      ToolConfid   
4748                  2           2626            IsConfid      ToolConfid   
7703                  0           4103            IsConfid      ToolConfid   
3131                  1           1789            IsConfid      ToolConfid   
4509                  3            969            IsConfid      ToolConfid   
1441                  6           3705            IsConfid      ToolConfid   
6249                  2           2617            IsConfid      ToolConfid   
2988                  6           3582            IsConfid      ToolConfid   

      Conversion  
721            1  
4564           1  
4397           1  
4748           1  
7703           0  
3131           1  
4509           1  
1441           1  
6249           1  
2988           1

# **Data Exploration**

## Define Data Information Function

In [3]:
def get_data_info(df):
    print("\n\033[1mShape of Dataframe:\033[0m ", df.shape)
    print("\n\033[1mColumns in Dataframe:\033[0m ", df.columns.to_list())
    print("\n\033[1mData Types of Columns:\033[0m\n", df.dtypes)

    print("\n\033[1mInformation About Dataframe:\033[0m")
    df.info()

    print("\n\033[1mNumber of Unique Values in Each Column:\033[0m")
    for col in df.columns:
        print(f"\033[1m{col}\033[0m: {df[col].nunique()}")
    
    print("\n\033[1mNumber of Null Values in Each Column:\033[0m\n", df.isnull().sum())

    print("\n\033[1mNumber of Duplicates Rows:\033[0m ", df.duplicated().sum())

    print("\n\033[1mDescriptive Stats of Dataframe:\033[0m\n", df.describe().transpose())

## Get Info

In [4]:
get_data_info(df)


Shape of Dataframe:  (8000, 20)

Columns in Dataframe:  ['CustomerID', 'Age', 'Gender', 'Income', 'CampaignChannel', 'CampaignType', 'AdSpend', 'ClickThroughRate', 'ConversionRate', 'WebsiteVisits', 'PagesPerVisit', 'TimeOnSite', 'SocialShares', 'EmailOpens', 'EmailClicks', 'PreviousPurchases', 'LoyaltyPoints', 'AdvertisingPlatform', 'AdvertisingTool', 'Conversion']

Data Types of Columns:
 CustomerID               int64
Age                      int64
Gender                  object
Income                   int64
CampaignChannel         object
CampaignType            object
AdSpend                float64
ClickThroughRate       float64
ConversionRate         float64
WebsiteVisits            int64
PagesPerVisit          float64
TimeOnSite             float64
SocialShares             int64
EmailOpens               int64
EmailClicks              int64
PreviousPurchases        int64
LoyaltyPoints            int64
AdvertisingPlatform     object
AdvertisingTool         object
Conversion      

# **Findings**

**What we find about the data**
* The data has 8.000 rows and 20 columns
* There are no duplicate and null values
* There are two confidential columns, AdvertisingPlatform & AdvertisingTool

**Some specific observation about the data**
* **Age**: Average age is around 43 years old, with a range of 18 to 69 years old.
* **Income**: There is a significant spread in income levels due to significant standard deviation with average income around $84.664
* **CTR**:
* **Conversion Rate**:
* **Website Visits**:

In [6]:
df = df.drop(['AdvertisingPlatform', 'AdvertisingTool'], axis=1)

# **Digital Marketing Campaign Success Metrics**

## **1. Click Through Rate (CTR)**
CTR is the number of clicks that ad receives divided by the number of times ad is shown: *clicks ÷ impressions = CTR*.

In [7]:
ctr = (df['ClickThroughRate'].sum() / len(df)) * 100
print(f"Click Through Rate (CTR): {ctr:.2f}%")

Click Through Rate (CTR): 15.48%


## **2. Conversion Rate**

Conversion rate is a percentage used in digital marketing to evaluate performance of website traffic, marketing campaigns and conversions. To calculate a conversion rate, take the number of *conversions divided by the total number of visitors*.

In [8]:
conversion_rate = (df['Conversion'].sum() / len(df)) * 100
print(f"Conversion Rate: {conversion_rate:.2f}%")

Conversion Rate: 87.65%


## **3. Customer Acquisition Cost (CAC)**

Customer Acquisition Cost (CAC) measures how much an organization spends to acquire new customers.

In [9]:
acquired_customers = len(df[df['Conversion'] == 1])
total_cost = df['AdSpend'].sum()
cac = total_cost / acquired_customers

print(f"Customer Acquisition Cost: ${cac:.2f}")

Customer Acquisition Cost: $5705.58


## **4. Marketing Cost by Campaign Channel**

In [10]:
mkt_cost = pd.pivot_table(df,
                          index='CampaignChannel', values='AdSpend',
                          aggfunc='sum', margins=True, margins_name='Total')
mkt_cost = mkt_cost.map(lambda x: '{:,.2f}'.format(x))
mkt_cost

AdSpend
CampaignChannel               
Email             7,871,575.85
PPC               8,199,236.98
Referral          8,653,518.69
SEO               7,740,903.88
Social Media      7,542,323.25
Total            40,007,558.64

## **5. Return on Ad Spend (ROAS)**

In this case, I assume the marketing cost is taken as 5% of the total income. This means, if the marketing cost incurred is 40,007,558.64, then the assumed income obtained is 800,151,173.

In [11]:
total_revenue = 800151173
roas = total_revenue / total_cost
print(f"Return on Ad Spend: {roas:.2f}")

Return on Ad Spend: 20.00


## **6. Return on Investment (ROI)**